File: modules/jooby-openapi/src/test/java/issues/i1905/Issue1905.java
package issues.i1905;

import static org.junit.jupiter.api.Assertions.assertEquals;

import io.jooby.openapi.OpenAPIResult;
import io.jooby.openapi.OpenAPITest;

public class Issue1905 {

  @OpenAPITest(value = App1905.class)
  public void shouldParseInstallApp(OpenAPIResult result) {
    assertEquals("openapi: 3.0.1\n"
        + "info:\n"
        + "  title: 1905 API\n"
        + "  description: 1905 API description\n"
        + "  version: \"1.0\"\n"
        + "paths:\n"
        + "  /sub:\n"
        + "    get:\n"
        + "      operationId: getSub\n"
        + "      responses:\n"
        + "        \"200\":\n"
        + "          description: Success\n"
        + "          content:\n"
        + "            application/json:\n"
        + "              schema:\n"
        + "                type: string\n"
        + "  /inline/sub:\n"
        + "    get:\n"
        + "      operationId: getInlineSub\n"
        + "      responses:\n"
        + "        \"200\":\n"
        + "          description: Success\n"
        + "          content:\n"
        + "            application/json:\n"
        + "              schema:\n"
        + "                type: string\n"
        + "  /lambda-body/sub:\n"
        + "    get:\n"
        + "      operationId: getLambdaBodySub\n"
        + "      responses:\n"
        + "        \"200\":\n"
        + "          description: Success\n"
        + "          content:\n"
        + "            application/json:\n"
        + "              schema:\n"
        + "                type: string\n"
        + "  /instance-reference/sub:\n"
        + "    get:\n"
        + "      operationId: getInstanceReferenceSub\n"
        + "      responses:\n"
        + "        \"200\":\n"
        + "          description: Success\n"
        + "          content:\n"
        + "            application/json:\n"
        + "              schema:\n"
        + "                type: string\n"
        + "  /static-reference/sub:\n"
        + "    get:\n"
        + "      operationId: getStaticReferenceSub\n"
        + "      responses:\n"
        + "        \"200\":\n"
        + "          description: Success\n"
        + "          content:\n"
        + "            application/json:\n"
        + "              schema:\n"
        + "                type: string\n", result.toYaml());
  }
}

File: modules/jooby-openapi/src/test/kotlin/kt/i1905/Issue1905.kt
package kt.i1905

import io.jooby.openapi.OpenAPIResult
import io.jooby.openapi.OpenAPITest
import org.junit.jupiter.api.Assertions

class Issue1905 {

  @OpenAPITest(value = App1905::class)
  fun shouldParseInstallApp(result: OpenAPIResult) {
    Assertions.assertEquals("""openapi: 3.0.1
info:
  title: 1905 API
  description: 1905 API description
  version: "1.0"
paths:
  /sub:
    get:
      operationId: getSub
      responses:
        "200":
          description: Success
          content:
            application/json:
              schema:
                type: string
  /static/ref/sub:
    get:
      operationId: getStaticRefSub
      responses:
        "200":
          description: Success
          content:
            application/json:
              schema:
                type: string
  /instance/ref/sub:
    get:
      operationId: getInstanceRefSub
      responses:
        "200":
          description: Success
          content:
            application/json:
              schema:
                type: string
  /supplier/sub:
    get:
      operationId: getSupplierSub
      responses:
        "200":
          description: Success
          content:
            application/json:
              schema:
                type: string
""", result.toYaml())
  }
}
